agenda/presentation/src/main/java/com/nibalk/tasky/agenda/presentation/components/AgendaCard.kt

```diff
@@ -0,0 +1,292 @@
+package com.nibalk.tasky.agenda.presentation.components
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.widthIn
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.MoreVert
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.rotate
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.nibalk.tasky.agenda.domain.AgendaItem
+import com.nibalk.tasky.core.presentation.components.TaskyCircularCheckbox
+import com.nibalk.tasky.core.presentation.themes.TaskyBrownLight
+import com.nibalk.tasky.core.presentation.themes.TaskyDarkGray
+import com.nibalk.tasky.core.presentation.themes.TaskyGreen
+import com.nibalk.tasky.core.presentation.themes.TaskyLightGreen
+import com.nibalk.tasky.core.presentation.themes.TaskyTheme
+import com.nibalk.tasky.core.presentation.themes.TaskyWhite
+import com.nibalk.tasky.core.presentation.themes.spacing
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+@Composable
+fun AgendaCard(
+    modifier: Modifier = Modifier,
+    item: AgendaItem,
+    onOptionsClick: () -> Unit,
+    onItemClick: () -> Unit,
+    onIsDone: (Boolean) -> Unit,
+) {
+    val itemDate by remember {
+        derivedStateOf {
```

Comment:
`derivedStateOf` is used when the keys change more often than the result which is not really the case here. Whenever the from or to date changes the formatted string also changes. In that case, the recommended approach is using `remember` with keys:
```
val itemDate = remember(item.eventFromDateTime, item.eventToDateTime) {
    // Format
}
```